java-squid/src/main/java/org/sonar/java/model/InternalSyntaxSpacing.java

```diff
@@ -17,28 +17,51 @@
  * License along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
  */
-package org.sonar.java.ast.visitors;
+package org.sonar.java.model;
 
-import com.google.common.base.Preconditions;
 import com.sonar.sslr.api.AstNode;
-import org.sonar.squidbridge.SquidAstVisitor;
-import org.sonar.squidbridge.api.SourceFile;
-import org.sonar.sslr.parser.LexerlessGrammar;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.TreeVisitor;
 
-import java.io.File;
+import java.util.Iterator;
 
-public class FileVisitor extends SquidAstVisitor<LexerlessGrammar> {
+public class InternalSyntaxSpacing extends JavaTree {
+
+  private final int start;
+  private final int end;
+
+  public InternalSyntaxSpacing(int start, int end) {
+    super(null);
+    this.start = start;
+    this.end = end;
+  }
 
   @Override
-  public void visitFile(AstNode astNode) {
-    File file = getContext().getFile();
-    getContext().addSourceCode(new SourceFile(file.getAbsolutePath(), file.getPath()));
+  public Kind getKind() {
+    //FIXME should have a dedicated kind associated with a dedicated interface.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)